spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -44,6 +46,31 @@
     import gettext
     _ = gettext.gettext
 
+
+def _find_pylintrc_path(path):
+    os.chdir(path)
+    return pylint.config.find_pylintrc()
+
+
+def get_pylintrc_path(search_paths=None, home_path=None):
+    """Get the path to the highest pylintrc file on a set of search paths."""
+    current_cwd = os.getcwd()
+    pylintrc_path = None
+    if home_path is None:
+        home_path = osp.expanduser("~")
+    try:
+        pylintrc_paths = [
+            _find_pylintrc_path(path) for path in search_paths if path]
+        pylintrc_path_home = _find_pylintrc_path(home_path)
+        for pylintrc_path in pylintrc_paths:
+            if (pylintrc_path is not None
+                    and pylintrc_path != pylintrc_path_home):
+                break
+    finally:

```

Comment:
Maybe should we add here an except statement to log the exception that could raise and properly show the kind of exception that could raise here?

Reply:
A fair point. We don't expect any particular exception here, this is only for safety so that the working directory is reset properly in case of an unexpected error, following the general principle of not modifying de-facto global state unless necessary. Two possible scenarios do come to mind: First, pylint not being installed, which would raise (and presumably handle, e.g. by displaying a user warning or disabling the plugin) many errors long before this in the code; the error would in fact the error would presumably occur on import of the pylint module, which would be the appropriate place to handle it if we want to if it is not handled with a warning message on startup, not here.

Second if some new major version of pylint were to move, rename or remove the `pylint.config.find_pylintrc()` function, which we _want_ to fail with an actual error, not just a subtle log message, so that we see it immediately on our CIs before users do as soon as such an update goes out and gets pulled in by Spyder, and if for whatever reason users _do_ see it first, it is handled by Spyder's normal error report dialog so it gets reported so we can see and fix it, not buried in a local log somewhere.